Observer/BackfillTransactions.php

```diff
@@ -15,309 +15,321 @@
     {
-        return $this->filterBuilder->setField('state')
-            ->setValue($state)
-            ->setConditionType('eq')
-            ->create();
+        $dataToEncode = [
+            'meta_information' => [
+                'orderIds' => $body,
+                'force' => (bool)$this->getInput('force'),
+            ],
+        ];
+
+        $data = [
+            'data' => [
+                'bulk_uuid' => $this->uuid,
+                'topic_name' => \Taxjar\SalesTax\Model\Configuration::TAXJAR_TOPIC_SYNC_TRANSACTIONS,
+                'serialized_data' => $this->serializer->serialize($dataToEncode),
+                'status' => \Magento\Framework\Bulk\OperationInterface::STATUS_TYPE_OPEN,
+            ]
+        ];
+
+        return $this->operationFactory->create($data);
     }
 
     /**
-     * @param $websiteId
-     * @return array
+     * @param \Magento\Store\Api\Data\StoreInterface $store
+     * @return bool
      */
-    protected function getWebsiteStoreIds($websiteId): array
+    private function compareStoreWebsiteId(\Magento\Store\Api\Data\StoreInterface $store): bool
     {
-        $storeIds = [];
-
-        foreach ($this->storeManager->getStores() as $store) {
-            if ($store->getWebsiteId() == $websiteId) {
-                $storeIds[] = $store->getId();
-            }
-        }
+        $websiteId = $store->getWebsiteId();
+        return (!is_null($websiteId) && $websiteId == $this->getInput('websiteId'));
+    }
 
-        return $storeIds;
+    /**
+     * @param \Magento\Sales\Api\Data\OrderInterface $order
+     * @return bool
+     */
+    private function isOrderSyncable(\Magento\Sales\Api\Data\OrderInterface $order): bool
+    {
+        return $order->getUpdatedAt() > $order->getTjSalestaxSyncDate();
     }
 
     /**
-     * @param AbstractModel $object
+     * @param \Magento\Sales\Model\AbstractModel $object
      * @return int|null
      */
-    protected function getEntityId($object): ?int
+    private function getEntityId(\Magento\Sales\Model\AbstractModel $object): ?int
     {
-        return $object->getEntityId();
+        return (int)$object->getEntityId();
     }
 
     /**
-     * @param string $bulkUuid
-     * @param $body
-     * @return OperationInterface
+     * @return array
      */
-    protected function makeOperation(
-        string $bulkUuid,
-        $body
-    ): OperationInterface {
-        $dataToEncode = [
-            'meta_information' => $body,
-        ];
-        $data = [
-            'data' => [
-                'bulk_uuid' => $bulkUuid,
-                'topic_name' => TaxjarConfig::TAXJAR_TOPIC_SYNC_TRANSACTIONS,
-                'serialized_data' => $this->serializer->serialize($dataToEncode),
-                'status' => \Magento\Framework\Bulk\OperationInterface::STATUS_TYPE_OPEN,
-            ]
+    private function getConfiguration(): array
+    {
```

Comment:
When triggering a sync from the UI I am getting this:

![image](https://user-images.githubusercontent.com/11321890/141857890-a5fafeeb-80cd-437b-ae77-c33ad04ebca6.png)

It appears it is not correctly getting the to and from parameters.

Update: on further investigation it looks like this is an issue with the get Input function.

Reply:
@dallendalton - Did you select "to" and "from" dates? I saw the same output when I didn't select any dates, but that was expected because the JSON is the user/system input, not the resulting search criteria used for the order lookup.

If you did select dates from the input fields and that was the output, then that is unexpected. I'll try it again on my local.

Reply:
@sethobey I selected the inputs. If no inputs are selected though we should still display the timeframe that was searched and not a null value. That way the logs are clear on exactly what search was performed.

Reply:
@dallendalton - This should be resolved now. I had renamed the expected AJAX param in PHP and updated tests accordingly, but missed it on the frontend.